lib/api.js

```diff
@@ -109,26 +109,12 @@ var api = {
    * }
    * ```
    */
-  messages: []
-}
-
-/**
- * Extends the tree with the it's API
- *
- * @module API
- *
- * @param   {Array} tree - PostHTML Tree
- *
- * @returns {Array} tree - PostHTML Tree (extended)
- */
-module.exports = function (tree) {
-  tree.walk = api.walk
-  tree.match = api.match
-  tree.messages = api.messages
+  this.messages = [].concat((Api.context && Api.context.messages) || [])
```

Comment:
move `messages` in PostHTML scope; Like that https://github.com/posthtml/posthtml/blob/master/lib/index.js#L101-L102

Reply:
```js
if (!result.apiExtended) {
  Api.call(result)
  result.messages = OldResult.messages // or messages in global
  result.options = OldResult.options // or options
  result.processor = _this // PostHTML 
}
```

Reply:
```js
function coreApi(old){
  Api.call(this)
  this.messages = old.messages // or messages in global
  this.options = old.options // or options
  this.processor = _this // PostHTML 
}
```
?

Reply:
I began to wonder what we are tying the process to api

Reply:
ok